File: calendar2/src/Service/ExtendedTimestamp.java
public class ExtendedTimestamp 
{

    public int Time;
    public String SenderIP;
    public int CriticalRegionNo;

    public ExtendedTimestamp(int time, String senderIP, int criticalRegionNo) 
    {
        Time = time;
        SenderIP = senderIP;
        CriticalRegionNo = criticalRegionNo;
    }

    public String ToString() 
    {
        return Time + "#" + SenderIP + "#" + CriticalRegionNo;
    }

    public static ExtendedTimestamp FromString(String timestampString) 
    {
        String[] splitArray = timestampString.split("#");

        if (splitArray.length != 3) {
            System.out.println("Timestamp decode failed!");
            return null;
        }

        return new ExtendedTimestamp(Integer.parseInt(splitArray[0]), splitArray[1], Integer.parseInt(splitArray[2]));
    }
}

File: calendar1/src/Service/ExtendedTimestamp.cs
using System;

namespace BitCalendarService
{
    class ExtendedTimestamp
    {
        public int      Time                { get; private set; }
        public string   SenderIP            { get; private set; }
        public int      CriticalRegionNo    { get; private set; }

        public ExtendedTimestamp(int time, string senderIP, int criticalRegionNo)
        {
            Time                = time;
            SenderIP            = senderIP;
            CriticalRegionNo    = criticalRegionNo;
        }

        public string ToString()
        {
            return Time + "#" + SenderIP + "#" + CriticalRegionNo;
        }
        
        public static ExtendedTimestamp FromString(string timestampString)
        {
            var splitArray = timestampString.Split(new[] {"#"}, StringSplitOptions.None);
            if (splitArray.Length != 3)
            {
                Console.WriteLine("Timestamp decode failed!");
                return null;
            }

            return new ExtendedTimestamp(int.Parse(splitArray[0]),splitArray[1],int.Parse(splitArray[2]));
        }
    }
}
